just_audio/android/src/main/java/com/ryanheise/just_audio/AudioPlayer.java

```diff
@@ -265,10 +265,13 @@ public void onPlayerError(ExoPlaybackException error) {
         errorCount++;
         if (player.hasNext() && currentIndex != null && errorCount <= 5) {
             int nextIndex = currentIndex + 1;
-            // TODO: pass in initial position here.
-            player.setMediaSource(mediaSource);
-            player.prepare();
-            player.seekTo(nextIndex, 0);
+            Timeline timeline = player.getCurrentTimeline();
+            if (nextIndex >= 0 && (timeline.isEmpty() || nextIndex < timeline.getWindowCount())) {
```

Comment:
Some of these conditions look a bit surprising:

* How can `nextIndex >= 0` be violated? For `nextIndex` to be -1, `currentIndex` would have to be -2 which would indicate a problem elsewhere.
* What is the scenario where `player.hasNext()` AND `timeline.isEmpty()`? If `timeline.isEmpty()`, what does `player.seekTo(nextIndex, 0)` do?

If all is reasonable, it would be good to add a comment above this condition to explain these surprises.

Reply:
yes it does look quit surprising, but this is `onPlayerError` method triggered so anything can happen here imho. Please take a look at this code from exoplayer:
https://github.com/google/ExoPlayer/blob/4b1e0fa9fc643f96ab5bbf4cec9aa7a79e4bc789/library/core/src/main/java/com/google/android/exoplayer2/ExoPlayerImpl.java#L610
I don't think that problem is in `nextIndex < 0` or `timeline.isEmpty()`. I guess the problem is that `nextIndex >= timeline.getWindowCount()` somehow.
Btw could you please provide a purpose of seeking player to beginning on every player error?

Reply:
I guess that is why the condition looks puzzling. In context, I don't think all of the terms of that condition are appropriate, because we have just computed the next index which should never be less than zero. It seems at least that condition could be removed, but maybe you can confirm it through debugging to check `nextIndex` is never less than zero. What do you think?

I guess the real mystery is why `hasNext` returns true and yet `nextIndex` then seems to be going out of range.

> Btw could you please provide a purpose of seeking player to beginning on every player error?

I believe this was to emulate the iOS behaviour where if you have a playlist and one item fails, it automatically advances to the next item.

Reply:
Although i cannot debug it (issue is not reproducible in normal circumstances) `nextIndex` seems fine for me. I guess the problem in `timeline`, maybe it has been reset to default values somehow right after player error.

Reply:
That's also my take. If we look at the ExoPlayer code you referenced, it looks like that `windowIndex` condition was only there to guard against an out of bound parameter value passed in by the client, but in our case `nextIndex` is not a parameter and we can guarantee it is within bounds. Can we remove the `nextIndex` condition?

Reply:
I'm also not sure about the use of the `timeline.isEmpty()` condition. If the timeline is empty, it means there is no "next" to play, and I think we should just do nothing. So just:

```dart
if (nextIndex < timeline.getWindowCount()) {
```